build/ci_utils/src/cache/goodie/graalvm.rs

```diff
@@ -166,14 +179,55 @@ mod tests {
     /// Check that we correctly recognize both the GraalVM version and the Java version.
     #[test]
     fn version_recognize() {
-        let version_string = r"openjdk version 17.0.6 2023-01-17
-OpenJDK Runtime Environment GraalVM CE 22.3.1 (build 17.0.6+10-jvmci-22.3-b13)
-OpenJDK 64-Bit Server VM GraalVM CE 22.3.1 (build 17.0.6+10-jvmci-22.3-b13, mixed mode, sharing)";
+        let version_string = "openjdk version \"17.0.7\" 2023-04-18
+OpenJDK Runtime Environment GraalVM CE 17.0.7+7.1 (build 17.0.7+7-jvmci-23.0-b12)
+OpenJDK 64-Bit Server VM GraalVM CE 17.0.7+7.1 (build 17.0.7+7-jvmci-23.0-b12, mixed mode, sharing)";
 
         let found_graal = graal_version_from_version_string(version_string).unwrap();
-        assert_eq!(found_graal, Version::new(22, 3, 1));
+        let expected_graal_version = Version {
+            major: 17,
+            minor: 0,
+            patch: 7,
+            pre:   Prerelease::EMPTY,
+            build: BuildMetadata::new("7.1").unwrap(),
+        };
+        assert_eq!(found_graal, expected_graal_version);
 
         let found_java = Java.parse_version(version_string).unwrap();
-        assert_eq!(found_java, Version::new(17, 0, 6));
+        assert_eq!(found_java, Version::new(17, 0, 7));
+    }
+
+    #[test]
+    fn recognize_oneline_version() {
+        let version_line =
+            "OpenJDK Runtime Environment GraalVM CE 17.0.7+7.1 (build 17.0.7+7-jvmci-23.0-b12)";
+        let graal_version = Version::find_in_text(version_line).unwrap();
+        let expected_graal_version = Version {
+            major: 17,
+            minor: 0,
+            patch: 7,
+            pre:   Prerelease::EMPTY,
+            build: BuildMetadata::new("7.1").unwrap(),
+        };
+        assert_eq!(graal_version, expected_graal_version);
+    }
+
+    #[test]
+    fn version_to_string() {
+        let version_with_build_metadata = Version {
+            major: 17,
+            minor: 0,
+            patch: 7,
+            pre:   Prerelease::EMPTY,
+            build: BuildMetadata::new("7.1").unwrap(),
+        };
+        let version_str = format!(
```

Comment:
As above. But then — I'm not sure if we need such test, as it'd end up testing `semver` library.

Reply:
Right, this test is just a leftover when I was fiddling with the versions. Removed in 0a0545ef71140d754ec539a5d6d1cc6cc9345bca